lib/about_page/configuration.rb

```diff
@@ -63,21 +64,35 @@ def set_headers! response
     end
 
     def to_h
-      self.hash.to_h.inject({}) { |h,v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
+      self.hash.to_h.inject({}) { |h, v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
     end
 
     def health_report
-      self.nodes.collect do |key, profile| 
-        if profile.class.validators.length > 0 
-          health = profile.valid? ? 'ok' : 'error'
-          errors = profile.errors.map {|a,m| "#{a} #{m}"}
-          { 'component' => key.to_s, 'status' => health, 'errors' => errors }
-        else
-          nil
+      self.nodes.collect do |key, profile|
+        next unless profile.class.validators.length.positive?
+
+        begin
+          health = Timeout.timeout(55) { profile.valid? ? 'ok' : 'error' }
+          errors = error_message(profile)
+        rescue Timeout::Error
+          health = 'error'
+          # ActiveModel::Errors.add expects the first argument to be the name
+          # of the method that generated the error. We do not have access to the
+          # name of the validator that timed out in this context, so define a
+          # dummy `timeout` method to reference when adding the error to the model.
+          profile.class.define_method(:timeout) {}
+          profile.errors.add(:timeout, message: ': component check has timed out.')
```

Comment:
I am explicitly passing a `message` argument here (and throughout the rest of the gem) for clarity but if the second argument of `add` is a string it gets used as the error message, so our configurations in [avalon-about](github.com/avalonmediasystem/avalon-about/lib/avalon/about) should still generate the errors properly.

A possible exception is our mediainfo configuration. The attr being passed into the errors class is `:mediainfo` which is not a method within the configuration and so I expect it will just trigger an ActiveModel error. We could potentially fix that by providing a dummy method like I do here or reworking the error generation in someway.